src/App.tsx

```diff
@@ -1,39 +1,69 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+type State = {
+  clockName: string,
+  hasClock: boolean,
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerID = 0;
+
+  componentDidMount() {
+    window.addEventListener('click', this.clockShow);
+    window.addEventListener('contextmenu', this.clockShow);
+
+    this.timerID = window.setInterval(() => {
+      this.setState(({ clockName: getRandomName() }
+      ));
+    }, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (this.state.hasClock === true
+      && this.state.clockName !== prevState.clockName) {
+      console.debug(`Renamed from ${prevState.clockName} to ${this.state.clockName}`);
+    }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    return prevProps;
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount() {
+    clearInterval(this.timerID);
+    window.addEventListener('click', this.clockShow);
+    window.addEventListener('contextmenu', this.clockShow);
+  }
 
-        {' time is '}
+  clockShow = (e: MouseEvent) => {
+    e.preventDefault();
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+    this.setState({ hasClock: e.button !== 2 });
```

Comment:
Unfortunately, it doesn't work. Did you test it? Ask questions in the chat if you need a hint, please